lib/src/barback.dart

```diff
@@ -34,7 +34,7 @@ import 'package:pub_semver/pub_semver.dart';
 /// constraint and barback's version.
 ///
 /// [compat]: https://gist.github.com/nex3/10942218
-final pubConstraints = {
+final pubConstraints = <String, VersionRange>{
```

Comment:
True. But in every case where you call in in this final, it is a `VersionRange`. Moreover, in every usage of `pubConstraints` you assume the type is `VersionRange`.

Seems better to fail here than elsewhere, no?